File: Backend/Microservices/Finance/src/main/java/tn/esprit/finance/entity/Budget.java
package tn.esprit.finance.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.AssertTrue;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.*;


import java.time.LocalDateTime;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Budget {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String projectId;  // Référence au projet

    @Positive(message = "Le montant total doit être un nombre positif")
    private double montantTotal;

    @PositiveOrZero(message = "Le montant restant ne peut pas être négatif")
    private double montantRestant;

    private LocalDateTime dateCreation;

    @AssertTrue(message = "Le montant total doit être supérieur au montant restant")
    public boolean isMontantTotalValid() {
        return montantTotal >= montantRestant;
    }
}

File: Frontend/FrontOffice_ConstructionIQ/src/app/models/budget.ts
export class Budget {
    id: number;
    projetId: number;
    montantTotal: number;
    montantRestant: number;
    dateCreation: string;
  
    constructor(id: number = 0, projetId: number = 0, montantTotal: number = 0, montantRestant: number = 0, dateCreation: string = '') {
      this.id = id;
      this.projetId = projetId;
      this.montantTotal = montantTotal;
      this.montantRestant = montantRestant;
      this.dateCreation = dateCreation;
    }
  }
  